accounts/middleware.py

```diff
@@ -25,16 +26,25 @@ def __call__(self, request):
                 body = {"token": token}
                 headers = {"Authorization": "Bearer {}".format(token)}
                 try:
-                    data = requests.post(
+                    platform_request = requests.post(
                         url=accounts_settings.PLATFORM_URL + "/accounts/introspect/",
                         headers=headers,
                         data=body,
                     )
-                    if data.status_code == 200:  # Access token is valid
-                        data = data.json()
-                        user = User.objects.filter(id=int(data["user"]["pennid"]))
-                        if len(user) == 1:  # User has an account on this product
-                            request.user = user.first()
+                    if platform_request.status_code == 200:  # Access token is valid
+                        json = platform_request.json()
+                        user_props = json["user"]
+                        # TODO: maybe update access token if a local one does not exist
+                        # user_props["token"] = {
+                        #     "access_token": token,
+                        #     "refresh_token": "",
+                        #     "expires_in": json["exp"] - int(timezone.now().timestamp()),
+                        # }
+                        user = auth.authenticate(remote_user=user_props, tokens=False)
+                        if user:  # User authenticated successfully
+                            request.user = user
+                        else:  # Error occurred
```

Comment:
still think this should be a gateway error, since at this point the credentials are valid but authenticate didn't work for some other reason

Reply:
Makes sense, will fix.